Makefile

```diff
@@ -8,16 +8,25 @@ ns ?= "test"
 
 # Determine the operating system
 UNAME_S := $(shell uname -s)
+PHP_API_VERSION := $(shell php -i | grep 'PHP API' | awk -F' => ' '{print $$2}')
+PHP_VERSION := $(shell php -v | head -n 1 | awk '{print $$2}' | cut -d. -f1,2)
+
+# Check if PHP version is greater than 8.0
+ifeq ($(shell awk 'BEGIN{ print ("$(PHP_VERSION)" >= "8.0") }'), 0)
+    $(error PHP version must be greater than or equal to 8.0)
+endif
+
+
 ifeq ($(UNAME_S),Darwin)
     EXTENSION := .dylib
-	EXT_DIR_PATH := /opt/homebrew/opt/php@8.1/pecl/20210902
-    PHP_INI_PATH := /opt/homebrew/etc/php/8.1/php.ini
-    RESTART_COMMAND := brew services restart php@8.1
+	EXT_DIR_PATH := /opt/homebrew/opt/php@$(PHP_VERSION)/pecl/$(PHP_API_VERSION)
+    PHP_INI_PATH := /opt/homebrew/etc/php/$(PHP_VERSION)/php.ini
+    RESTART_COMMAND := brew services restart php@$(PHP_VERSION)
 else ifeq ($(UNAME_S),Linux)
     EXTENSION := .so
-	EXT_DIR_PATH := /usr/lib/php/20210902/
-    PHP_INI_PATH := /etc/php/8.1/cli/php.ini
-    RESTART_COMMAND := sudo systemctl restart php8.1-fpm && sudo systemctl restart apache2
+	EXT_DIR_PATH := /usr/lib/php/$(PHP_API_VERSION)/
```

Comment:
Same here. Ubuntu/Debian/Fedora/Arch may have different installation paths. Have you checked them all?

Reply:
The most reliable is to use `php-config --extension-dir`, which requires php-devel on RHEL/Fedora, and similar devel-only packages on others. A fallback using only php cli is `php -r 'echo ini_get("extension_dir");'`

Reply:
Thanks @thias this works!